api/libproject/devstart.rb

```diff
@@ -1589,38 +1589,56 @@ def export_workspace_data(cmd_name, *args)
     :fn => ->(*args) {export_workspace_data("export-workspace-data", *args)}
 })
 
-def generate_impersonated_user_token(cmd_name, *args)
+def generate_impersonated_user_tokens(cmd_name, *args)
   common = Common.new
   ensure_docker cmd_name, args
 
   op = WbOptionsParser.new(cmd_name, args)
   op.add_typed_option(
-      "--output-token-filename [output-token-filename]",
+      "--output-token-filenames [output-token-filename1, ...]",
       String,
-      ->(opts, v) { opts.output_token_filename = v},
-      "Path to an output file for the generated token")
+      ->(opts, v) { opts.output_token_filenames = v},
+      "Comma-separated paths to output file(s) for the generated token(s)")
   op.add_typed_option(
-      "--impersonated-username [impersonated-username]",
+      "--impersonated-usernames [impersonated-username1, ...]",
       String,
-      ->(opts, v) { opts.impersonated_username = v},
-      "AoU researcher email to impersonate, e.g. calbach@fake-research-aou.org")
-  op.add_validator ->(opts) { raise ArgumentError unless (opts.output_token_filename and opts.impersonated_username)}
+      ->(opts, v) { opts.impersonated_usernames = v},
+      "Comma-separated AoU researcher email(s) to impersonate, e.g. calbach@fake-research-aou.org")
+  op.add_validator ->(opts) { raise ArgumentError unless (opts.output_token_filenames and opts.impersonated_usernames)}
   op.parse.validate
 
+  user_email_domain = nil
+  op.opts.impersonated_usernames.split(',').each do |username|
+   split_email = username.split('@')
+   if split_email.nil? || split_email[1].nil?
+     raise ArgumentError.new("Username is not a valid email address: " + username)
+   end
+
+   if user_email_domain.nil?
+     user_email_domain = split_email[1]
+   else
+     if user_email_domain != split_email[1]
+       raise ArgumentError.new("All usernames must have the same email domain. #{split_email[1]} does not match #{user_email_domain}")
+     end
+   end
+  end
+
+  # derive the project_id from the usernames, failing if this is not possible
   project_id = nil
   ENVIRONMENTS.each_key do |project|
     if project == "local"
       next
     end
 
     config = get_config(project)
-    if op.opts.impersonated_username.end_with?("@" + config["googleDirectoryService"]["gSuiteDomain"])
+    if user_email_domain == config["googleDirectoryService"]["gSuiteDomain"]
```

Comment:
Since we're doing this check anyways, it probably simplifies the above code if we grab the domain from config first, then just verify that all of the input user emails end in `@[domain]`.

Reply:
The config is actually the last thing we get: email -> domain -> project -> config.

We could retrieve the config with the first user and then compare the others to that.  Not sure if that is an improvement.

Reply:
Ah sorry I misread this. I thought project was a parameter. Ignore this comment.